argus/scenarios/service_mock.py

```diff
@@ -268,3 +271,36 @@ def x509():
             TO9FwPXIRcR/XrcMDn7slaHtbILM3P4pIbaXUhvel+qLOAMp6k2iDl2Q
             -----END CERTIFICATE-----
             """.strip())
+
+
+class HTTPKeysMetadataServiceApp(BaseServiceApp):
+    """Custom OpenStack http metadata."""
+
+    METADATA = {
+        "keys": [
+            {
+                "name": "argus_cert",
+                "type": "x509",
+                "data": util.get_certificate()
+            }
+        ] + [{
+            "name": "argus_key",
+            "type": "ssh",
+            "data": data
+        } for data in util.get_public_keys()]
+    }
+
+    def _get_metadata(self):
+        """Fill-in the metadata password provided by the config file."""
+        metadata = self.METADATA.copy()
+        key = "admin_pass"
+        metadata[key] = self.scenario.get_metadata()[key]
+        return metadata
```

Comment:
Move METADATA here, transform _get_metadata in a property with util.cached_property and fill the password explicitly in that dict.